File: src/main/java/axirassa/model/PingerEntity.java

package axirassa.model;

import java.io.Serializable;
import java.util.Set;

import javax.persistence.Basic;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

import lombok.Getter;
import lombok.Setter;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;

import axirassa.util.AutoSerializingObject;

@Entity
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
public class PingerEntity extends AutoSerializingObject implements Serializable, EntityWithUser {
	private static final long serialVersionUID = -6709719920544228167L;


	public static String createBroadcastQueueName (Long userId, Long pingerId) {
		return "ax.account." + userId + ".pinger." + pingerId;
	}


	//
	// INSTANCE
	//

	// ID
	@Id
	@Getter
	@Setter
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;


	// USER
	@Getter
	@Setter
	@ManyToOne(fetch = FetchType.LAZY, optional = false)
	private UserEntity user;


	// URL
	@Getter
	@Setter
	@Basic(optional = false)
	private String url;


	/**
	 * The number of seconds between checks
	 */
	@Getter
	@Basic(optional = false)
	private int frequency; 

	public void setFrequency(int frequency) {
		this.frequency= frequency;
	}

	public void setFrequency (PingerFrequency frequency) {
		this.frequency = frequency.getInterval();
	}


	@Getter
	@Setter
	@Basic(optional = false)
	@Enumerated(EnumType.STRING)
	public MonitorType monitorType;
}

File: src/main/java/axirassa/model/PingerEntity.scala
package axirassa.model

import axirassa.util.AutoSerializingObject
import org.hibernate.annotations.Type
import javax.persistence.Basic
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import scala.reflect.BeanProperty
import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy

import javax.persistence.ManyToOne
import javax.persistence.FetchType
@Entity
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
class PingerEntity extends EntityWithUser with AutoSerializingObject {
    @Id
    @BeanProperty
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id : Long = _

    @BeanProperty
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    var user : UserEntity = _

    @BeanProperty
    @Basic(optional = false)
    var url : String = _

    @BeanProperty
    @Basic(optional = false)
    var frequency : Int = _

    @BeanProperty
    @Basic(optional = false)
    @Type(`type` = "axirassa.model.MonitorType")
    var monitorType : Enumeration = _
}

object PingerEntity {
    def createBroadcastQueueName(userId : Long, pingerId : Long) = {
        "ax.account."+userId+".pinger."+pingerId
    }
}